recipes/metaknowledge/meta.yaml

```diff
@@ -0,0 +1,47 @@
+{% set name = "metaknowledge" %}
+{% set version = "3.3.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: f137dce7ed7e1fb3a73730716d37e2b5757d9fc94f3bb50f8e82338c38b17dae
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```

Comment:
It looks like you need `entry_points`, see these links:
 * https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html?highlight=selectors#python-entry-points
 * https://github.com/networks-lab/metaknowledge/blob/master/setup.py#L45-L49
(this is probably causing the Windows failure)

Reply:
Ah yes, thanks. I'm not too familiar with `entry_points` and the interaction with `conda`. Presumably, while building the wheel using `pip` the `entry_point` is created, I presume? If I include the `entry_point` also in the `meta.yaml`, will the problem then be resolved?

Reply:
Yeah, unfortunately this information has to be duplicated between the `setup.py` and `meta.yaml` file. So far as I know, it is used for two things:
 * Making the binary for the `entry_point` relocatable
 * Installing the correct binary for the current platform for `noarch` packages, i.e. if it was built on Linux then installed on Windows it will install `my_entry_point.exe` instead of `my_entry_point`.

Reply:
OK, great, thanks! 

No problem to have some duplication of this, just wanted to make sure that I would do the right thing.

Reply:
I believe the new commit should resolve this problem.

Reply:
Right, there cannot be selectors when using `noarch`. How would you suggest how to resolve this, @chrisburr? The `pyreadline` is only available (and necessary) for Windows. Should I just remove the `noarch`? 

FYI, the `(py)readline` is only necessary because of the CLI that is created using the `entry_point`. I don't think the CLI is necessary in `conda-forge`. Is there a way to remove it from the `conda` package? I could simply exclude the CLI (`metaknowledge.bin`) in the test section, but the CLI would still not function properly with `pyreadline`.

Reply:
> Should I just remove the noarch?

Yes, in this case there isn't an alternative. You will also need to remove ` >=3` on `python` and use `skip: True. # [py2k]` instead.

Also `pyreadline [win]` -> `pyreadline  # [win]`

> Is there a way to remove it from the conda package?

You could patch it, but I think its better to be consistent else users might get confused why the same version behaves differently between `pip` and `conda`.

Reply:
You could patch it if it is not that important and the recipe will still be ``norach``

Reply:
I would go to patch the ``entry_points`` and remove the dependency of ``pyreadline ``

Reply:
I tend to agree with @chrisburr, it is probably best to ensure that users get the same versions. Personally, I consider the CLI unimportant, but perhaps other users may think differently about this. Thanks for the advice!